homeassistant/components/workday/binary_sensor.py

```diff
@@ -165,6 +174,23 @@ async def async_setup_entry(
         _holiday_string = holiday_date.strftime("%Y-%m-%d")
         LOGGER.debug("%s %s", _holiday_string, name)
 
+    async def check_date(service_call: ServiceCall) -> ServiceResponse:
+        """Check if date is workday or not."""
+        date_to_test: date = service_call.data[CHECK_DATE]
+        holiday_date = date_to_test in obj_holidays
+
+        date_string = date_to_test.strftime("%Y-%m-%d")
+
+        return {"date": date_string, "workday": not holiday_date}
```

Comment:
```suggestion
        return {"workday": not holiday_date}
```

The service is called with the date already. It makes no sense to return the date, or what do you think?

Reply:
It will be removed